content/english/java-tictactoe/activity-5.md

```diff
@@ -1,7 +1,111 @@
 ---
 title: "5. (Optional) AI Move"
-description: ""
+description: "Use Minimax algorithm to pick move for the computer"
 date: 2020-07-23T00:00:00Z
 weight: 6
-hidden: true
----
\ No newline at end of file
+difficulty: "Intermediate"
+hidden: false
+---
+{{% notice tip %}}
+#### This is a bonus activity!
+To continue, you should have completed the Tic-Tac-Toe program from the previous exercises! 
+#### Prerequisites (from Java: Basics) 
+- activity-1: <a href="../../java-basics/activity-1" target="_blank">Print Statements and Comments</a>
+- activity-2: <a href="../../java-basics/activity-2" target="_blank">Variables and Types</a>
+- activity-3: <a href="../../java-basics/activity-3" target="_blank">Operators</a>
+- activity-4: <a href="../../java-basics/activity-4" target="_blank">Control Structures</a>
+- activity-5: <a href="../../java-basics/activity-5" target="_blank">Methods</a>
+- activity-6: <a href="../../java-basics/activity-6" target="_blank">Object and Classes</a>
+- activity-7: <a href="../../java-basics/activity-7" target="_blank">Data Structures (Array & ArrayList)</a>
+{{% /notice %}}
+
+### Let's make the computer think
+Previously, we use a `Random` object to randomly generate a move for the computer. As a result, the competitiveness of the computer is fairly low.
+
+In this exercise, we want to increase the difficulty of the game by making optimal decisions for the computer.
+
+We will do so by adding artificial intelligence into our program with the use the <b>Minimax</b> algorithm (a well-defined procedure that allows computers to solve problems).
```

Comment:
I feel there will be questions like why here - like why are we looking at so many moves, etc, and how etc. It will be nice to mention something like the following:

through our code, we will be looking into the future and calculating the score of all final states to make a decision at any given point. While it is difficult to do it mentally when we are playing the game ourselves, computers can do this very easily using something called _'recursion'_
Note that by giving scores to states, we are teaching our code what a win is, and how to reach that score..